Auth0/IDTokenValidator.swift

```diff
@@ -45,9 +45,14 @@ struct IDTokenValidator: JWTAsyncValidator {
     }
 
     func validate(_ jwt: JWT, callback: @escaping (LocalizedError?) -> Void) {
-        signatureValidator.validate(jwt) { error in
-            if let error = error { return callback(error) }
-            callback(self.claimsValidator.validate(jwt))
+        DispatchQueue.global(qos: .userInitiated).async {
+            self.signatureValidator.validate(jwt) { error in
```

Comment:
Any potential issues around retain cycles?

Reply:
We actually want to keep a strong reference to `self`, since we later need it to execute the claims validator. Using `[weak self]` would not increase the ref count and therefore risks deallocation before the claims validator gets called (specially since there's a network request involved for signature validation, that could really delay that moment), because the validator itself is not referenced anywhere. Using `[unowned self]` would pose the same problem plus a runtime crash. 

AFAIK a strong reference to `self` here would not cause a strong reference cycle because the result callback is merely passed around before ultimately being executed, and is not referenced anywhere.

Reply:
@hzalaz thoughts?

Reply:
And since the validator is a struct, it's actually a value type and not a reference type. It will be copied.